server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java

```diff
@@ -150,50 +149,46 @@ public MonitorScheduler getMonitorScheduler(
   @Provides
   @ManageLifecycle
   public JvmMonitor getJvmMonitor(
-      DataSourceTaskIdHolder dataSourceTaskIdHolder
+      TaskHolder taskHolder
   )
   {
-    Map<String, String[]> dimensions = MonitorsConfig.mapOfDatasourceAndTaskID(
-        dataSourceTaskIdHolder.getDataSource(),
-        dataSourceTaskIdHolder.getTaskId()
+    Map<String, String[]> dimensions = MonitorsConfig.mapOfTaskHolderDimensions(
+        taskHolder
     );
     return new JvmMonitor(dimensions);
   }
 
   @Provides
   @ManageLifecycle
   public JvmCpuMonitor getJvmCpuMonitor(
-      DataSourceTaskIdHolder dataSourceTaskIdHolder
+      TaskHolder taskHolder
   )
   {
-    Map<String, String[]> dimensions = MonitorsConfig.mapOfDatasourceAndTaskID(
-        dataSourceTaskIdHolder.getDataSource(),
-        dataSourceTaskIdHolder.getTaskId()
+    Map<String, String[]> dimensions = MonitorsConfig.mapOfTaskHolderDimensions(
+        taskHolder
     );
     return new JvmCpuMonitor(dimensions);
   }
 
   @Provides
   @ManageLifecycle
-  public JvmThreadsMonitor getJvmThreadsMonitor(DataSourceTaskIdHolder dataSourceTaskIdHolder)
+  public JvmThreadsMonitor getJvmThreadsMonitor(TaskHolder taskHolder)
   {
-    Map<String, String[]> dimensions = MonitorsConfig.mapOfDatasourceAndTaskID(
-        dataSourceTaskIdHolder.getDataSource(),
-        dataSourceTaskIdHolder.getTaskId()
+    Map<String, String[]> dimensions = MonitorsConfig.mapOfTaskHolderDimensions(
+        taskHolder
```

Comment:
Nit: Can we put this (and other similar invocations of this new method) in a single line?